src/parsec_service/stability.md

```diff
@@ -0,0 +1,68 @@
+# Parsec Stability
+
+Work in progress, see the [related issue](https://github.com/parallaxsecond/parsec-book/issues/83).
+
+## Why is stability important?
+
+Parsec is a long-running service that will be configured, deployed, upgraded and maintained over a
+long period of time. The environment will also have client applications that are calling it, and
+persistent state that is collected on storage. Components will be upgraded over time. For Parsec to
+be supportable in enterprise system, it must not be brittle against upgrades. If Parsec is upgraded
+and restarted, client applications should continue working, and all persistent state from prior
+versions should remain valid and usable.
+
+## What does stability mean for the Parsec service?
+
+The following definition of stability is chosen for the Parsec service: *whatever two versions A and
+B of the Parsec binary, B being newer than A, A and B are stable if A can be removed and replaced by
+B without breaking anything*.
+
+Note that we are not looking at stability in the reverse direction: B can not be replaced by A
+without breaking anything.
+
+The principle of [semantic versioning](https://semver.org/) is used to describe the stability of
+Parsec versions and when breaking changes are done. If `parsec` version is at `1.0.0` then all
+future stable version to that will be `1.x.y`.
+
+## What needs to be stable?
+
+With the above definition, Parsec stays stable if it keeps the exact same communication channels and
+format with its environment. Those are already described in the [dataflow
+diagram](../parsec_security/parsec_threat_model/threat_model.md#dataflow-diagram) in the threat
+model.
+
+The communication channels that needs to be stable are (some of them are not in the diagram):
+
+1. Communication with clients
+2. Communication with authenticators
+3. Communication received from the CLI invocation
+4. Configuration file (including default configuration options)
+5. Key mappings
```

Comment:
Fair enough!

Reply:
Actually, I think it is better to limit the document to what is there in Parsec today. If there is a need in the future for the service to store other kind of persistent data, then we should address the stability of that separately. The reason is that the methods we are going to use to store the key mappings in a stable way might be different to what we use for other kind of data.

Reply:
I'm not sure that I agree - this feels like letting details get in the way of the goal. If the aim of this stability document is to describe what we mean by stability for the service, then I think we need scope that in terms of all possible persistent state that might be stored and read back later. This doesn't preclude having a clear statement of how this would be achieved for individual pieces like the key mappings - I just think that we need a very strong top-level statement that covers all avenues.

Reply:
> then I think we need scope that in terms of all possible persistent state that might be stored and read back later.

I see your thinking. I also think that this document should describe what we mean by stability for the service. I guess my methodology here was to list all current and used communication channels as opposed as the ones we could/will have in the future. That way for each of them we can actually link a specific issue to do the work to make sure stability is enforced and tested.

I am happy to modify this item as you suggest though: a top-level "Internally stored persistent states" statement and then indivual items like key mappings and the ones we add in the future.